R/create_dap_specific_concept.R

```diff
@@ -24,107 +24,112 @@
 #' @export
 
 create_dap_specific_concept <- function(codelist, data_db, name_attachment, save_db, date_col_filter = NULL,
-                                        table_name = 'cdm_table_name',
-                                        column_name_prefix = 'column_name',
-                                        expected_value_prefix = 'expected_value',
+                                        table_name = "cdm_table_name",
+                                        column_name_prefix = "column_name",
+                                        expected_value_prefix = "expected_value",
                                         add_meaning = FALSE) {
-  
   if (nrow(codelist) <= 0) {
     stop("Codelist does not contain any data.")
   }
-  
-  
+
+
   # Get unique tables from codelist
   scheme <- unique(codelist[[table_name]])
   # Get columns and value names
-  cols_names <- grep(paste0("^",column_name_prefix), names(codelist), value = TRUE)
-  value_names <- grep(paste0("^",expected_value_prefix), names(codelist), value = TRUE)
-  # Get columns and values
-  cols <- codelist[, ..cols_names]
-  values <- codelist[, ..value_names]
-  
-  # Preprocess all possible tables:
-  # Loop through each table in scheme
-  for (name in scheme) {
-    # Edit table name to indicate that it's been processed
-    name_edited <- paste0(name, "_EDITED")
-    # Get columns that need to be converted to uppercase
-    to_upper_cols <- na.omit(unique(unlist(codelist[get(table_name) %in% name, ..cols_names])))
-    # Get all columns from the table
-    columns_db_table <- DBI::dbListFields(data_db, name)
-    # Get columns that don't need to be converted to uppercase
-    rest_cols <-  na.omit(columns_db_table[!columns_db_table %in% to_upper_cols])
-    # Create query to convert columns to uppercase
-    to_upper_query <- paste0(paste0("UPPER(", to_upper_cols, ") AS ", to_upper_cols), collapse = ", ")
-    # Create query to select all columns and that are not converted to uppercase
-    select_cols_query <- paste0(paste0(rest_cols, collapse = ", "), " ,")
-    # If edited table doesn't exist in save_db, create a temporary table
-    if (!name_edited %in% DBI::dbListTables(save_db)) {
-      rs <- DBI::dbSendStatement(save_db, paste0("CREATE TEMP TABLE ", name_edited, " AS
+  cols_names <- grep(paste0("^", column_name_prefix), names(codelist), value = TRUE)
+  value_names <- grep(paste0("^", expected_value_prefix), names(codelist), value = TRUE)
+  if (length(cols_names) == 0 | length(value_names) | length(cols_names) != length(value_names)) {
```

Comment:
should it be ...length(value_names) **== 0** ... ?

Reply:
Yes! you are right, fixed.